libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackClientWrapper.cs

```diff
@@ -519,13 +521,14 @@ public virtual async Task<AuthTestResponse> TestAuthAsync()
         /// <param name="ts">The timestamp of the message.</param>
         /// <param name="channelId">The channel to delete the message from.</param>
         /// <param name="text">The text to update with.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
         /// <param name="botName">The optional bot name.</param>
         /// <param name="parse">Change how messages are treated.Defaults to 'none'. See https://api.slack.com/methods/chat.postMessage#formatting. </param>
         /// <param name="linkNames">If to find and link channel names and username.</param>
         /// <param name="attachments">The attachments, if any.</param>
         /// <param name="asUser">If the message is being sent as user instead of as a bot.</param>
         /// <returns>A <see cref="UpdateResponse"/> representing the response to the operation.</returns>
-        public virtual async Task<UpdateResponse> UpdateAsync(string ts, string channelId, string text, string botName = null, string parse = null, bool linkNames = false, Attachment[] attachments = null, bool asUser = false)
+        public virtual async Task<UpdateResponse> UpdateAsync(string ts, string channelId, string text, CancellationToken cancellationToken = default(CancellationToken), string botName = null, string parse = null, bool linkNames = false, Attachment[] attachments = null, bool asUser = false)
```

Comment:
CancellationToken should be always the last parameter of the signature.

Reply:
Only when not optional, surely

Reply:
No, always. If there are other optional parameters, CancellationToken should be at the end too.

Reply:
good to know